frontend/lib/src/dataframes/Quiver.ts

```diff
@@ -875,6 +876,26 @@ but was expecting \`${JSON.stringify(expectedIndexTypes)}\`.
       .format(timeInSeconds % 1 === 0 ? "HH:mm:ss" : "HH:mm:ss.SSS")
   }
 
+  private static formatDecimal(value: Uint32Array, scale: number): string {
+    // This code is partly based on: https://github.com/apache/arrow/issues/35745
```

Comment:
I think the complexity of this code is not explained in the inspiration, and this is some complex logic that could benefit a little explanation. Is it possible to explain the purpose of the lines and perhaps have it walk through an example in each comment?

Reply:
Yep, sounds good 👍  I added a couple of comments